### PR TITLE
feat: export component types

### PR DESCRIPTION
**Pull request by m0ksem**
Expose component prop types

### CODE REVIEW

**Comment by m0ksem** [comment]
@RVitaly1978, look at this commit https://github.com/epicmaxco/vuestic-ui/pull/1896/commits/01d42947107bb73952287e2196174b4d444d43a1 as an example of component prop type expose.

File: packages/ui/src/components/va-accordion/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 import withConfigTransport from '../../services/config-transport/withConfigTransport'
-import VaAccordion from './VaAccordion.vue'
+import _VaAccordion from './VaAccordion.vue'
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-accordion/index.ts:2]:
```suggestion
import VaAccordion as VaAccordionType from './VaAccordion.vue'
```

File: packages/ui/src/components/va-progress-bar/VaProgressBar.vue [hunk 2]
```diff
@@ -96,7 +96,7 @@ export default defineComponent({
         animationDirection: props.reverse ? 'reverse' : 'normal',
       })),
 
-      ariaValueNowComputed: computed(() => props.indeterminate ? null : props.modelValue),
+      ariaValueNowComputed: computed(() => props.indeterminate ? undefined : props.modelValue),
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-progress-bar/VaProgressBar.vue:99]: Isn't the default value `null`?

File: packages/ui/src/components/va-select/VaSelect.vue [hunk 3]
```diff
@@ -148,26 +148,21 @@
 <script lang="ts">
 import { defineComponent, PropType, ref, computed, watch, nextTick, Ref } from 'vue'
 
-import { useSelectableList, useSelectableListProps, SelectableOption } from '../../composables/useSelectableList'
+import { useSelectableList, useSelectableListProps, useSelectablePropsFn } from '../../composables/useSelectableList'
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-select/VaSelect.vue:151]:
```suggestion
import { useSelectableList, useSelectablePropsFn } from '../../composables/useSelectableList'
```

File: packages/ui/src/composables/useSelectableList.ts [hunk 4]
```diff
@@ -14,6 +14,16 @@ export const useSelectableListProps = {
...
+export const useSelectablePropsFn = <O extends SelectableOption, StOrFn = StringOrFn<O>>() => ({
+  options: { type: Array as PropType<O[]>, default: () => [] },
+  textBy: { type: [String, Function] as PropType<StOrFn>, default: 'text' },
+  valueBy: { type: [String, Function] as PropType<StOrFn>, default: '' },
+  trackBy: { type: [String, Function] as PropType<StOrFn>, default: 'value' },
+  disabledBy: { type: [String, Function] as PropType<StOrFn>, default: 'disabled' },
+  groupBy: { type: [String, Function] as PropType<StOrFn>, default: 'group' },
+})
+
 export function useSelectableList (props: ExtractPropTypes<typeof useSelectableListProps>) {
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useSelectableList.ts:27]: What about `useSelectableListProps` here?

File: packages/ui/src/components/va-counter/VaCounter.demo.vue [hunk 5]
```diff
@@ -282,10 +282,10 @@
...
 <script>
 import { ref } from 'vue'
-import VaCounter from './index'
-import VaButton from './../va-button'
-import VaChip from './../va-chip'
-import VaIcon from './../va-icon'
+import { VaCounter } from './index'
+import { VaButton } from './../va-button'
+import { VaChip } from './../va-chip'
+import { VaIcon } from './../va-icon'
```
[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/src/components/va-counter/VaCounter.demo.vue:288]: What do you think about?
```ts
import { VaButton, VaChip, VaIcon } from '..'
```

File: packages/ui/src/components/va-data-table/types.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  classes?: DateTableClassesOptions
+  headerClasses?: DateTableClassesOptions
+  style?: DateTableStyleOptions
+  headerStyle?: DateTableStyleOptions
+}
+
+export type DateTableColumnSource = DateTableColumn | string
+
+// inner representation of the columns
+export interface TableColumn {
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-data-table/types.ts:31]:
```suggestion
export interface DataTableColumnInnerImplementation {
```
something like this

File: packages/ui/src/components/va-toast/VaToast.vue [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<template>
+ToastPosition<template>
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-toast/VaToast.vue:1]:
```suggestion
<template>
```

File: packages/ui/src/components/va-counter/VaCounter.demo.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/components/va-counter/VaCounter.demo.vue]: This can be handy since we're exporting a named component. But I prefer explicit imports

**Review by RVitaly1978 — APPROVED** [review]

**Review by Derranion — APPROVED** [review]

**Pull request merged**